nats.go

```diff
@@ -2754,7 +2754,9 @@ func (nc *Conn) SubscribeSync(subj string) (*Subscription, error) {
 	mch := make(chan *Msg, nc.Opts.SubChanLen)
 	s, e := nc.subscribe(subj, _EMPTY_, nil, mch)
 	if s != nil {
+		s.mu.Lock()
```

Comment:
I would not do that. Instead, I would either let nc.subscribe() infer the type, or if not possible, pass the type as an argument. We need to have the type set before the subscription is added to the connection's subs map.

Reply:
ok will take another look.

Reply:
Take a look now..